### DISCUSSION TITLE
Q: 数据迁移助手从Artalk迁移到任何一种存储服务都无法工作 || Q: Data Migration Assistant does not work when migrating from Artalk to any storage service

### DISCUSSION THREAD

**Discussion by BeaCox** [walinejs/waline#2023] [Q&A] [answered]
### Describe the bug | 问题描述

控制台报错
```
framework-0d72886b.js:1 TypeError: Cannot read properties of undefined (reading 'replace')
    at MigrationTool-6dbcee3c.js:1:923
    at Array.map (<anonymous>)
    at I (MigrationTool-6dbcee3c.js:1:823)
    at wsql (MigrationTool-6dbcee3c.js:4:777)
    at v (MigrationTool-6dbcee3c.js:16:450)
    at ot (framework-0d72886b.js:1:13328)
    at Se (framework-0d72886b.js:1:13406)
    at HTMLButtonElement.n (framework-0d72886b.js:1:58094)
```

### Website URL | 问题网站

https://waline.js.org/migration/tool.html

### Where your waline deploy? | 服务部署在哪里？

Deta

### Where your comment data store? | 数据存储在哪里？

MySQL

---
### Describe the bug | Problem description

console error
```
framework-0d72886b.js:1 TypeError: Cannot read properties of undefined (reading 'replace')
    at MigrationTool-6dbcee3c.js:1:923
    at Array. map (<anonymous>)
    at I (MigrationTool-6dbcee3c.js:1:823)
    at wsql (MigrationTool-6dbcee3c.js:4:777)
    at v (MigrationTool-6dbcee3c.js:16:450)
    at ot (framework-0d72886b.js:1:13328)
    at Se (framework-0d72886b.js:1:13406)
    at HTMLButtonElement.n (framework-0d72886b.js:1:58094)
```

### Website URL | Problem Website

https://waline.js.org/migration/tool.html

### Where your waline deploy? | Where is the service deployed?

Deta

### Where your comment data store? | Where is your comment data store?

MySQL

**Comment by lizheming** [comment]
提供一下导入的数据 demo 吧，后面我测一下

---
Provide a demo of the imported data, I will test it later

**Reply by BeaCox** [reply]
email你了，请查收，感谢！

---
email you, please check, thanks!

**Comment by lizheming** [comment]
https://github.com/walinejs/waline/commit/c325758752e9285686eb8fa078c5fdb42a2d337c 已对 Artalk 新数据结构进行了适配，可以再试试

---
https://github.com/walinejs/waline/commit/c325758752e9285686eb8fa078c5fdb42a2d337c has adapted the new data structure of Artalk, you can try again

**Reply by BeaCox** [reply]
谢谢，但是在后台导入时报错：
![07_13_2311.png](https://s2.loli.net/2023/07/13/anlbpVs5MUDQjKd.png)
迁移至leancloud/cloudbase/mysql.../github都试过了，都有报错。
我使用的是deta部署


---
Thank you, but an error is reported when importing in the background:
![07_13_2311.png](https://s2.loli.net/2023/07/13/anlbpVs5MUDQjKd.png)
I have tried migrating to leancloud/cloudbase/mysql.../github, but all of them reported errors.
I am using a deta deployment

**Reply by lizheming** [reply, marked as answer]
迁移助手导出的文件，需要进 LeanCloud/cloudbase 后台进行导入，不是在 Waline 后台导入的，Waline 后台只接受 Waline 后台导出的格式。

---
The files exported by the Migration Assistant need to be imported into the LeanCloud/cloudbase background, not the Waline background. The Waline background only accepts the format exported by the Waline background.

**Reply by BeaCox** [reply]
好的，感谢！另外twikoo的迁移似乎也有些问题

---
ok thanks! In addition, the migration of twikoo seems to have some problems

**Reply by lizheming** [reply]
提供 demo 数据看下

---
Provide demo data to see

**Reply by BeaCox** [reply]
已经邮件你了！

---
Already emailed you!

**Reply by lizheming** [reply]
已更新，可以再试试

---
It has been updated, you can try again

**Reply by BeaCox** [reply]
测试无误，感谢！

---
The test is correct, thanks!
